### DISCUSSION TITLE
Server support

### DISCUSSION THREAD

**Discussion by erikdoe** [erikdoe/ccmenu2#10] [General]
Please use this discussion to express interest in which servers CCMenu should support. Please use emojis to upvote existing proposals. There's no guarantee that the most-voted server type will be supported first. Having a publicly accessible test server greatly increases the chances of future support.

The servers in [this list](https://cctray.org/servers/) support the cctray format and should be compatible wth CCMenu. Many of them have their own API's which provide more information than the standard tray feed. So, specific support could still make sense in some cases.

**Comment by brunns** [comment]
GitHub Actions

**Reply by erikdoe** [reply]
Supported since version 20.

**Comment by brunns** [comment]
Jenkins

**Reply by erikdoe** [reply]
Supported via the cctray standard. Are there specific features in Jenkins' own API that you would like to see supported?

**Reply by brunns** [reply]
Not especially - the cctray standard is fine AFAIC.

**Comment by brunns** [comment]
AWS CodeBuild

**Comment by brunns** [comment]
AWS CodePipeline

**Comment by brunns** [comment]
TeamCity

**Comment by maneesh29s** [comment]
Gitlab CI

**Reply by maneesh29s** [reply]
As of now this is the most upvoted request.

Since Gitlab CI does not support CCtray specifications, specific solution might be required.

So are there any plans to this?

**Reply by erikdoe** [reply]
Yes. I have verfied that I can create a working pipeline on a free acount at gitlab.com. The current plan is to finish up some of the existing issues (#11, #14, #22) and make a release with those. Then I'll likely have a look at GitLab next.

**Reply by maneesh29s** [reply]
That's good to know

**Reply by sven-agnew** [reply]
I'm also watching this project and in particular this area of work with interest as we use Gitlab CI as well.

**Reply by erikdoe** [reply]
Finally, I've started work on GitLab support (see #27). Please watch that issue if you're interested. I'll likely make a few prereleases for testing and would appreciate feedback.

**Reply by maneesh29s** [reply]
Great, thank you @erikdoe  for picking this up!

**Reply by erikdoe** [reply]
I've now finished (I think!) all the work that needs to be done for CCMenu to work with gitlab.com. Please check out the developer release here: https://github.com/ccmenu/ccmenu2/releases

**Reply by erikdoe** [reply]
I would also be interested whether this already works for self-hosted GitLab. You can override the URLs using the Mac's defaults systems like this:
```
defaults write net.net.sourceforge.cruisecontrol.CCMenu GitLabBaseURL <URL-for-installation>
defaults write net.net.sourceforge.cruisecontrol.CCMenu GitLabAPIBaseURL <URL-for-API>
```
The API URL probably includes a path, e.g. `https://gitlab.com/api/v4`

**Comment by rodrigodealer** [comment]
CircleCI

**Reply by erikdoe** [reply]
CircleCI is listed as supporting the cctray format: https://cctray.org/servers/

Is there a particular feature that you'd like to see supported? Something that CircleCI makes available via the API?

**Reply by rodrigodealer** [reply]
I tried with different ways setting up the server with CCMenu, but I'm not sure if the API's are still supported, given I received a lot of 404's.

**Reply by GentlemanHal** [reply]
@rodrigodealer Bit late, but both of the CCTray URLs listed at https://cctray.org/servers/ for CircleCI are still working correctly for me in general. So the CCTray format is still currently supported by CircleCI, although it isn't mentioned anywhere in the docs anymore so wouldn't be surprised if it gets stealth dropped at some point.

**Comment by earloc** [comment]
Azure DevOps

**Reply by erikdoe** [reply]
If I were to implement and test this with a free plan (https://azure.microsoft.com/en-us/pricing/details/devops/azure-devops-services/), that would work, right?

**Reply by earloc** [reply]
yepp, should work (at least for the Cloud/SaaS).

There is still the "Server" - Version, but this will retire somewhen in the near future.

So the API should work with other organizations / Private projects.

**Comment by barryw** [comment]
Drone CI

I run a self-hosted drone server which uses github auth. It does support the cctray xml file, but doesn't work with ccmenu's basic auth. Its API does support bearer token auth which works with the cctray file, so I'm working on a PR to add bearer token auth. I have it working and just need to add tests. I think others may benefit from this.
